MSAL/src/native_auth/controllers/reset_password/MSALNativeAuthResetPasswordController.swift

```diff
@@ -266,14 +266,20 @@ final class MSALNativeAuthResetPasswordController: MSALNativeAuthBaseController,
                            context: context,
                            format: "Error in resetpassword/challenge request (resend code) \(error.errorDescription ?? "No error description")")
             return .init(.error(error: error, newState: nil))
-        case .redirect,
-                .unexpectedError:
+        case .redirect:
             let error = ResendCodeError()
             stopTelemetryEvent(event, context: context, error: error)
             MSALLogger.log(level: .error,
                            context: context,
                            format: "Error in resetpassword/challenge request (resend code) \(error.errorDescription ?? "No error description")")
             return .init(.error(error: error, newState: nil))
+        case .unexpectedError(let message):
```

Comment:
Here I had to split the code even though the difference is only in the `let message` because otherwise the compiles throws an error. If there is any way around this let me know. 
I could capture the code below the line `let error = ResendCodeError()` into a function as it's replicated but that would add ambiguity in my opinion

Reply:
the only way I see to don't duplicate this code is declaring a variable (example: `var message: String? = nil`) before the switch block, merge the two cases and instead of declaring a new variable for the unexpectedError case, you reuse the variable declared before (`case .redirect, .unexpectedError(message):`).